Automattic/jetpack
# Social: Add image validation for Instagram connections

Instagram connections will work a bit different compared to other Publicize connections. It needs an image to be uploaded, so if a user is on a free plan, the featured image will be used, if they have the Advanced plan the image will be picked from SIG, picked image, featured image, in this order.

To avoid bad requests the image needs to be validated, which is handled by this PR.

## Proposed changes:

* Added restrictions for instagram into our restrictions array
* Created a hook that validates the featured image for a given array of connections
* Added a notice and disabled instagram connection if there is no image, or if it's the featured image and it is invalid

<img width="282" alt="CleanShot 2023-05-15 at 16 09 05 png" src="https://github.com/Automattic/jetpack/assets/36671565/9afd7cca-033f-4e37-a922-6011b93e12c2">

### Other information:


## Jetpack product discussion


1204479765121257-as-1204528424246673
## Does this pull request change what data or activity we track or use?


no.
## Testing instructions:






* It's a bit hard to test this before we have the instagram connection. The best way is to change `const CONNECTIONS_NEED_MEDIA = [ 'instagram' ];` in `projects/js-packages/publicize-components/src/components/form/index.js` for facebook.
* Add a Facebook connection to your site

On a free plan:
* The connection should be turned off by default. Adding a featured image re-enables it.
* If the featured image if bigger than the limit - which is now FB-s limit and is 4MB, the connection should be disabled as before

On the Advanced plan:
* The connection should be on by default. Disabling SIG in the post, so that you have no featured image, no picked image either should result in the connection being disabled and the notice appears. Adding any of the three should re-enable the connection
